shared/runtime/softtimer.c

```diff
@@ -30,11 +30,15 @@
 #include "py/runtime.h"
 #include "softtimer.h"
 
-#define TICKS_PERIOD 0x80000000
-#define TICKS_DIFF(t1, t0) ((int32_t)(((t1 - t0 + TICKS_PERIOD / 2) & (TICKS_PERIOD - 1)) - TICKS_PERIOD / 2))
-
 extern __IO uint32_t uwTick;
 
+STATIC inline int32_t ticks_diff(uint32_t t1, uint32_t t0) {
+    // t1 is after t0 (i.e. positive result) if there exists a uint32_t X < INT_MAX
```

Comment:
I think it should be `X <= INT_MAX`.

`INT_MAX` is `(1<<31)-1`, which is positive.  So, eg, we can have `t0=0` and `t1=INT_MAX` and the result of this function is positive.

Reply:
I made this change